### ISSUE TITLE
TreeGrid - invisible nodes with editor set after a few open/close clicks

### ISSUE THREAD

**Issue by pellmann** [vaadin/flow#8846]
With the given code below nodes (below the clicked one) will be come invisible after the node "deep.two" is opened/closed two times. 

This only happens with the editor setted:

`column.setEditorComponent(new TextField());`

Tested with Vaadin 14.3.3, Java 11.0.6, OSX 10.15.6, Newest Chrome and with Spring Boot 2.2.0

`import com.vaadin.flow.component.Component;
import com.vaadin.flow.component.grid.contextmenu.GridContextMenu;
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.html.Span;
import com.vaadin.flow.component.textfield.TextField;
import com.vaadin.flow.component.treegrid.TreeGrid;
import com.vaadin.flow.data.provider.hierarchy.AbstractHierarchicalDataProvider;
import com.vaadin.flow.data.provider.hierarchy.HierarchicalDataProvider;
import com.vaadin.flow.data.provider.hierarchy.HierarchicalQuery;
import com.vaadin.flow.function.ValueProvider;

import java.util.ArrayList;
import java.util.List;
import java.util.Optional;
import java.util.stream.Stream;

public class TestTree extends TreeGrid<TestTree.TestNode> {

    public TestTree() {
        setSizeFull();

        TestNode root = new TestNode(null, "root");
        root.addChild(new TestNode(root, "first"));
        TestNode second = root.addChild(new TestNode(root, "second"));
        second.addChild(new TestNode(second, "second.first"));
        second.addChild(new TestNode(second, "second.second"));
        TestNode secondThird = second.addChild(new TestNode(second, "second.third"));
        secondThird.addChild(new TestNode(secondThird, "deep.one"));
        TestNode deepTwo = secondThird.addChild(new TestNode(secondThird, "deep.two"));
        TestNode deepThree = secondThird.addChild(new TestNode(secondThird, "deep.three"));
        TestNode deepFour = secondThird.addChild(new TestNode(secondThird, "deep.four"));
        deepFour.addChild(new TestNode(deepFour, "last.child"));
        deepTwo.addChild(new TestNode(deepTwo, "between.one"));
        deepTwo.addChild(new TestNode(deepTwo, "between.two"));
        deepThree.addChild(new TestNode(deepTwo, "between.two.one"));
        deepThree.addChild(new TestNode(deepTwo, "between.two.two"));
        deepThree.addChild(new TestNode(deepTwo, "between.two.three"));

        // set data
        setDataProvider(new TestHierarchyProvider(root));

        // set view
        Column<TestNode> column = addComponentHierarchyColumn((ValueProvider<TestNode, Component>) TestNode::getView);

        // BUG: as soon as there is a an editor componenet set, the third time open/close on "deep.two" creates invisible nodes under it
        column.setEditorComponent(new TextField());

        // to remove
        GridContextMenu<TestNode> contextMenu = new GridContextMenu<>();
        contextMenu.setTarget(this);
        contextMenu.addItem("Remove", listener -> {
            Optional<TestNode> optionalItem = listener.getItem();
            optionalItem.ifPresent(node -> {
                TestNode parent = node.getParent();
                node.detach();
                getDataProvider().refreshItem(parent, true);
            });
        });

    }

    static public class TestNode {

        private final List<TestNode> children = new ArrayList<>();
        private TestNode parent;
        private String name;

        public TestNode(TestNode parent, String name) {
            this.parent = parent;
            this.name = name;
        }

        public String getName() {
            return name;
        }

        public void setName(String name) {
            this.name = name;
        }

        public Component getView() {
            Div div = new Div();
            div.add(new Span("."), new Span(name), new Span("."));
            return div;
        }

        public List<TestNode> getChildren() {
            return children;
        }

        public TestNode addChild(TestNode child) {
            children.add(child);
            return child;
        }

        public void detach() {
            getParent().getChildren().remove(this);
            parent = null;
        }

        public TestNode getParent() {
            return parent;
        }

    }

    public static class TestHierarchyProvider
            extends AbstractHierarchicalDataProvider<TestNode, String>
            implements HierarchicalDataProvider<TestNode, String> {

        private final TestNode root;

        public TestHierarchyProvider(TestNode root) {
            this.root = root;
        }

        @Override
        public int getChildCount(HierarchicalQuery<TestNode, String> query) {
            TestNode parent = query.getParent();
            if (parent == null) {
                return 1;
            }
            return parent.getChildren().size();
        }

        @Override
        public Stream<TestNode> fetchChildren(HierarchicalQuery<TestNode, String> query) {
            TestNode parent = query.getParent();
            if (parent == null) {
                return Stream.of(root);
            }
            return parent.getChildren().stream();
        }

        @Override
        public boolean hasChildren(TestNode item) {
            return !item.getChildren().isEmpty();
        }

        @Override
        public boolean isInMemory() {
            return true;
        }
    }

}`
